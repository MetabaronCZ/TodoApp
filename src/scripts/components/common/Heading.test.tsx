import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Heading } from 'components/common/Heading';
import { withMockedProviders } from 'test/component';

describe('components/common/Heading', () => {
  it('should render correctly', () => {
    const tree = render(withMockedProviders(<Heading>CONTENT</Heading>));
    expect(tree.container).toMatchSnapshot();
    expect(tree.getByRole('heading')).toBeInTheDocument();
    expect(tree.getByText('CONTENT')).toBeInTheDocument();
  });

  it('should render props', () => {
    const tree = render(withMockedProviders(<Heading tag="h3" />));
    expect(tree.container).toMatchSnapshot();
    expect(tree.container.querySelector('h3')).toBeInTheDocument();
  });
});
