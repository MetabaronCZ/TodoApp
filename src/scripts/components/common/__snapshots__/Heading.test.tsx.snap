// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Heading should render correctly 1`] = `
.c0 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 18px;
  line-height: 30px;
  font-weight: normal;
  margin: 0;
}

<div>
  <h1
    class="c0"
  >
    CONTENT
  </h1>
</div>
`;

exports[`components/common/Heading should render props 1`] = `
.c0 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 18px;
  line-height: 30px;
  font-weight: normal;
  margin: 0;
}

<div>
  <h3
    class="c0"
  />
</div>
`;
